### PR TITLE
feat(workspace): add user group resource support

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:
add user group resource support
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add user group resource support
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/workspace" TESTARGS="-run TestAccUserGroup_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccUserGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccUserGroup_basic
=== PAUSE TestAccUserGroup_basic
=== CONT  TestAccUserGroup_basic
--- PASS: TestAccUserGroup_basic (84.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 84.921s

```

### CODE REVIEW

File: docs/resources/workspace_user_group.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+* `name` - The name of user.
+
+* `email` - The email of user.
+
+* `phone` - The phone of user.
+
+* `description` - The description of user.
+
+* `total_desktops` - The number of desktops the user has.
```
Comment (Lance52259) [inline docs/resources/workspace_user_group.md:70]:
```suggestion
* `users` - The user information under the user group.
  The [users](#workspace_users_attr) structure is documented below.

<a name="workspace_users_attr"></a>
The `users` block supports:

* `name` - The name of user.

* `email` - The email of user.

* `phone` - The phone of user.

* `description` - The description of user.

* `total_desktops` - The number of desktops the user has.
```
The level of these attributes is not right

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_user_group.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+						"description":    user.Description,
+						"total_desktops": user.TotalDesktops,
+					})
+				}
+				mErr = multierror.Append(mErr, d.Set("users", users))
+			}
+
+			return diag.FromErr(mErr.ErrorOrNil())
+		}
+	}
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_user_group.go]:
```suggestion
	for _, userGroup := range userGroups {
		groupID := userGroup.ID
		if groupID == d.Id() {
			continue
		}
		...
	}
```

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_user_group.go [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+	if err != nil {
+		return diag.Errorf("error creating Workspace user group: %s", err)
+	}
+	d.SetId(createOpt.Name)
+	refreshIdDiags := refreshUserGroupID(client, d)
+	var diags diag.Diagnostics
+	diags = append(diags, refreshIdDiags...)
+
+	users := d.Get("users").([]interface{})
+	if len(users) > 0 {
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_user_group.go]: if the user group ID not found, the error does not throw at the first time, and keep running code logics.
An error will be throwed and warning you the group ID not exist from the user request API.

File: docs/resources/workspace_user_group.md (hunk 1 above)
Comment (jinyangyang222) [reply docs/resources/workspace_user_group.md:70]: fixed.

**Pull request merged**